spec/unit/models/thread.spec.ts

```diff
@@ -503,11 +503,59 @@ describe("Thread", () => {
                 const userId = "user1";
                 const { thread } = await createThreadAndEvent(client, 200, 100, userId);
 
-                // Then no receipt was added to the thread (the receipt is still for the thread root).
+                // Then no receipt was added to the thread (the receipt is still
+                // for the thread root). This happens because since we have no
+                // recursive relations support, we know that sometimes events
+                // appear out of order, so we have to check their timestamps as
+                // a guess of the correct order.
                 expect(thread.getReadReceiptForUserId(userId)?.eventId).toEqual(thread.rootEvent?.getId());
             });
         });
 
+        describe("With relations recursion support", () => {
+            it("Creates a local echo receipt for new events", async () => {
+                // Assumption: no server side support because if we have it, events
+                // can only be added to the timeline after the thread has been
+                // initialised, and we are not properly initialising it here.
+                expect(Thread.hasServerSideSupport).toBe(FeatureSupport.None);
+
+                // Given a client WITH relations recursion support
+                const client = createClientWithEventMapper(
+                    new Map([[Feature.RelationsRecursion, ServerSupport.Stable]]),
+                );
```

Comment:
This could go to a `beforeEach` block because it is shared code.

Reply:
I hate beforeEach blocks https://www.artificialworlds.net/blog/2013/03/27/setup-and-teardown-considered-harmful/ :-)